src/quicklookUI.cs

```diff
@@ -172,5 +173,69 @@ interface QLPreviewingController {
 		[Mac (10,15)]
 		[Export ("preparePreviewOfFileAtURL:completionHandler:")]
 		void PreparePreviewOfFile (NSUrl url, Action<NSError> completionHandler);
+
+		[iOS (15,0), Mac (12,0), MacCatalyst (15,0)]
+		[Export ("providePreviewForFileRequest:completionHandler:")]
+		void ProvidePreview (QLFilePreviewRequest request, Action<QLPreviewReply, NSError> handler);
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
```

Comment:
add `[DisableDefaultCtor]`

the API seems to be used only inside a protocol and that means the instance is created by the caller (OS) and not by user code

Another hint is that there's no way to set the` FileUrl` property which makes user-created instance not _really_ usable.

Reply:
All your comments address things I also missed in QuickLook - I'll make a separate PR to correct those in that framework https://github.com/xamarin/xamarin-macios/pull/12337/files

Reply:
https://github.com/xamarin/xamarin-macios/pull/12402